go/private/actions/archive.bzl

```diff
@@ -196,6 +196,7 @@ def emit_archive(go, source = None, _recompile_suffix = "", recompile_internal_d
         _validation_output = out_nogo_validation,
         _nogo_fix_output = out_nogo_fix,
         _cgo_deps = cgo_deps,
+        _headers = depset([f for f in source.srcs if f.path.split(".")[-1].lower().startswith("h")]),
```

Comment:
From a maintainability perspective, it would be better if these could be wrapped with `cc_common.create_compilation_context` and added into `cgo_deps` so that we only have a single place where headers are tracked and added from. 

But if that's too complicated or doesn't work, the current approach would be fine too

Reply:
Makes sense, forget my idea in that case :⁠-⁠)

Some non-Go files may be consumed locally (think `.s`), so that may add too much. We can start with `.h` and expand as long as we keep the attribute private, as you did.

@jayconrod Do you see a problem with this (or a better approach)?

Reply:
So these .h files are Go assembly headers, not C/C++/regular assembly headers, right? I don't really remember how these work, but it seems sensible to include them in `GoArchive` regardless of whether cgo is enabled. We can't really distinguish them from C header files, but maybe that's fine.

Not sure I get why they should be transitive though. Is it because headers from dependencies need to include other headers from their dependencies? If so, the test should demonstrate that.

Reply:
Transitive was my first guess. I figured they are not "direct" in that the go package depends on the compiled go archive. I'm not sure what the implications are with direct vs transitive in this case.

Reply:
For a given package, all the files in that package needed to produce the compiled archive are typically direct dependencies. If a file is needed to build other packages that import that package, then it could go into the transitive set.

Go typically avoids transitive file dependencies, since they can cause O(n^2) growth. The main use case is linking the final binary.

I'm not sure about assembly headers though. You mentioned `github.com/cloudflare/circl`. Is there a specific package within that module we should look at?

Reply:
In package [circl/dh/x25519](https://pkg.go.dev/github.com/cloudflare/circl@v1.6.0/dh/x25519), this [assembler file](https://github.com/cloudflare/circl/blob/main/dh/x25519/curve_amd64.s#L7) makes use of a header file defined in package [circl/math/fp25519](https://github.com/cloudflare/circl/blob/main/math/fp25519/fp_amd64.h).

Reply:
In my testing, this change alone fixed circl